src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/_params.py

```diff
@@ -112,21 +112,33 @@ def load_arguments(self, _):
         c.ignore('filter')
     # endregion
 
-    # region Diagnostics
     with self.argument_context('monitor diagnostic-settings') as c:
-        c.argument('resource_uri', options_list=['--resource-id'])
+        c.argument('name', options_list=('-n', '--name'))
+
+    with self.argument_context('monitor diagnostic-settings show') as c:
+        c.resource_parameter_context('resource_uri', required=True, arg_group='Target Resource')
+
+    with self.argument_context('monitor diagnostic-settings list') as c:
+        c.resource_parameter_context('resource_uri', required=True)
+
+    with self.argument_context('monitor diagnostic-settings delete') as c:
+        c.resource_parameter_context('resource_uri', required=True, arg_group='Target Resource')
+
+    with self.argument_context('monitor diagnostic-settings update') as c:
+        c.resource_parameter_context('resource_uri', required=True, arg_group='Target Resource')
 
     with self.argument_context('monitor diagnostic-settings create') as c:
-        c.argument('target_resource_id', options_list=['--resource-id'], validator=validate_diagnostic_settings)
+        c.resource_parameter_context('resource_uri', required=True, arg_group='Target Resource', skip_validator=True)
         c.argument('logs', type=get_json_object)
         c.argument('metrics', type=get_json_object)
-        c.argument('tags', nargs='*')
+        c.argument('event_hub_name')
+        c.argument('event_hub_rule_id')
```

Comment:
What are these two entries accomplishing?  There are no kwargs in the argument context to apply here, and no kwargs in the argument calls themselves.

Reply:
will remove.